pennylane/ops/qubit/matrix_ops.py

```diff
@@ -480,49 +479,83 @@ def compute_decomposition(D: TensorLike, wires: WiresLike) -> list["qml.operatio
-        ops = [QubitUnitary(qml.math.tensordot(global_phase, qml.math.eye(2), axes=0), wires[0])]
-        for wire0 in range(n):
-            # Single PauliZ generators correspond to the coeffs at powers of two
-            ops.append(qml.RZ(coeffs[1 << wire0], wires[n - 1 - wire0]))
-            # Double PauliZ generators correspond to the coeffs at the sum of two powers of two
-            ops.extend(
-                qml.IsingZZ(
-                    coeffs[(1 << wire0) + (1 << wire1)],
-                    [wires[n - 1 - wire0], wires[n - 1 - wire1]],
-                )
-                for wire1 in range(wire0)
-            )
+        **Finding the parameters:**
 
-        # Add all multi RZ gates that are not generated by single or double PauliZ generators
-        ops.extend(
-            qml.MultiRZ(c, [wires[k] for k in np.where(term)[0]])
-            for c, term in zip(coeffs, product((0, 1), repeat=n))
-            if sum(term) > 2
-        )
-        return ops
+        Theorem 7 referenced above only tells us the structure of the circuit, but not the
+        parameters for the ``SelectPauliRot`` and ``DiagonalQubitUnitary`` in the decomposition.
+        In the following, we will only write out the diagonals of all gates.
+        Consider a ``DiagonalQubitUnitary`` on :math:`n` qubits that we want to decompose:
+
+        .. math::
+
+            D(\theta) = (\exp(i\theta_0), \exp(i\theta_1), \dots,
+            \exp(i\theta_{N-2}), \exp(i\theta_{N-1})).
+
+        Here, :math:`N=2^n` is the Hilbert space dimension for :math:`n` qubits, which is
+        the same as the number of parameters in :math:`D`.
+
+        A ``SelectPauliRot`` gate using ``RZ`` rotations, or multiplexed ``RZ`` rotation, using the
+        first :math:`n-1` qubits as controls and the last qubit as target, takes the form
+
+        .. math::
+
+            UCR_Z(\phi) = (\exp(-\frac{i}{2}\phi_0), \exp(\frac{i}{2}\phi_0), \dots,
+            \exp(-\frac{i}{2}\phi_{N/2-1}), \exp(\frac{i}{2}\phi_{N/2-1})),
+
+        i.e., it moves the phase of neighbouring pairs of computational basis states by
+        the same amount, but in opposite direction. There are :math:`N/2` parameters
+        in this gate.
+        Similarly, a ``DiagonalQubitUnitary`` acting on the first :math:`n-1` qubits only (the
+        ones that were controls for ``SelectPauliRot``) takes the form
+
+        .. math::
+
+            D'(\theta') = (\exp(i\theta'_0), \exp(i\theta'_0), \dots,
+            \exp(i\theta'_{N/2-1}), \exp(i\theta'_{N/2-1})).
+
+        That is, :math:`D'` moves the phase of neighbouring pairs of basis states by the same
+        amount and in the same direction. It, too, has :math:`N/2` parameters.
+        Now, we see that we can compute the rotation angles,
+        or phases, :math:`\phi` and :math:`\theta'` quite easily from the original :math:`\theta`:
+
+        .. math::
+
+            (\exp(i\theta_{2i}), \exp(i\theta_{2i+1})) &\overset{!}{=}
```

Comment:
Is this a common notation? I don't know what that means 
\overset{!}{=}

Reply:
Done.